Bio/Seq.py

```diff
@@ -137,7 +144,7 @@ def __hash__(self):
         particular) as this has changed in Biopython 1.65. Older versions
         would hash on object identity.
         """
-        return hash(str(self))
+        return hash(bytes(self))
```

Comment:
This worried me, but apparently the hash of strings and bytes agree, at least for ASCII:

```pycon
>>> hash(b"ACGT")
5056019133776406303
>>> hash("ACGT")
5056019133776406303
```